### DISCUSSION TITLE
repseqio

### DISCUSSION THREAD

**Discussion by MilaBrun** [milaboratory/mixcr#1084] [Q&A]
Hi there, 

I have a question regarding the custom reference library design. It is recommended on the Website to use repseqio to generate JSON formatted libraries, but the link to the repository doesn't work. 
Is there an alternative tool to format the library ?

Then I also have a question regarding the usage of repseqio (I was lucky to install it already 6 month ago, so it's running on my computer). 
Every step works fine for my reference genome, except the last step where one "compiles" the library. 
It always give me an error message that repseqio was not able to "get the sequence for embedded...."
I tried to use the "not compiled" library in mixcr, bit it did not work. 

Does anyone know what's the reason for this and how can I fix this error ?

Thank you in advance, 
Mila

**Comment by Alex-Davydov** [comment]
Hi Mila,

Here is the link for the latest repseqio build:
[https://cdn.milaboratory.com/software/repseqio/repseqio-1.8.0-27-master-1.8.0-27-master.zip](url)
Regarding the error, the fasta files your library was build from should be in the same directory as JSON file.

Hope this helps,
Alex

**Comment by MilaBrun** [comment]
Hi Alex, 

thank you for your response. 

Yes there are in the same folder and I generated the library following the instructions on mixcr 
https://mixcr.com/mixcr/guides/create-custom-library/#using-the-library
Everything works fine for me except the last step to compile the library, I also tried it with having only the corresponding fasta files in a separate folder, and in the mixcr library folder...but same error message. 

I don't really know what to try else ?

Cheers, 
Mila

**Reply by Alex-Davydov** [reply]
Hi Mila,
Could you attach here the whole error message and example of your fasta file.
Best,
Alex

**Reply by MilaBrun** [reply]
Hi Alex, 

here is the error message: 
Exception in thread "main" java.lang.RuntimeException: Can't get sequence for embedded://801d133589fd4db18fd9ba54fc2a5790/IGHD1 (rel. /Users/MB/repseqio)
	at com.milaboratory.core.sequence.provider.CachedSequenceProvider$NoProvider.forceInitialize(CachedSequenceProvider.java:257)
	at com.milaboratory.core.sequence.provider.CachedSequenceProvider.forceInitialize(CachedSequenceProvider.java:141)
	at io.repseq.cli.CompileAction.compile(CompileAction.java:80)
	at io.repseq.cli.CompileAction.go(CompileAction.java:51)
	at com.milaboratory.cli.JCommanderBasedMain.main(JCommanderBasedMain.java:155)
	at io.repseq.cli.Main.main(Main.java:91)

and here is the corresponding fasta file: 
[IGHD_BroilerV4.fasta.zip](https://github.com/milaboratory/mixcr/files/11371409/IGHD_BroilerV4.fasta.zip)

Thank you in advance, 
Mila

**Reply by Alex-Davydov** [reply]
Hi Mila,
It seems you need to specify the output fasta file (unpadded fasta that library refers to). Here is the command for the attached fasta file:
<pre><code>repseqio fromPaddedFasta -t 9031 -c IGH -s broiler -j 3 -n 0 -g D -PDBegin=0 -PDEnd=-1 IGHD_BroilerV4.fasta <b>IGHD_Broiler.fasta</b> my_library.d.json -f</code></pre>
Best,
Alex

**Comment by PoslavskySV** [comment]
Hi,

now we recommend using MiXCR `buildLibrary` command: https://mixcr.com/mixcr/guides/create-custom-library/
